Utils.go

```diff
@@ -22,17 +24,36 @@ func SortUtxos(utxos []UTxO.UTxO) []UTxO.UTxO {
 	return res
 }
 
+type txIn struct {
```

Comment:
This seems way over complicated, unless there's something I'm missing...


```
func SortInputs(inputs []UTxO.UTxO) []UTxO.UTxO {
  sortedInputs := append(nil, inputs...)
  sort.Slice(sortedInputs, func(i, j int) bool {
    iTxId := hex.EncodeToString(sortedInputs[i].Input.TransactionId)
    jTxId := hex.EncodeToString(sortedInputs[j].Input.TransactionId)
    if iTxId != jTxId {
      return iTxId < jTxId
    } else {
      return sortedInputs[i].Input.Index < sortedInputs[j].Input.Index
    }
  }
  return sortedInputs
}
```

Reply:
Yeah, that should work and is obviously better. I was focused on fixing the bug, so I didn't realize that it was possible to simplify.

Reply:
ok no worries :sweat_smile: I was just nervous about the introduced throwaway struct